### ISSUE TITLE
Question: Progress in NER CFR Reference and support for AllenNLP GRU-NER

### ISSUE THREAD

**Issue by loretoparisi** [stanfordnlp/CoreNLP#677]
According to the [docs](https://nlp.stanford.edu/software/CRF-NER.html) the reference paper of the current CRF tagger was

> Jenny Rose Finkel, Trond Grenager, and Christopher Manning. 2005. Incorporating Non-local Information into Information Extraction Systems by Gibbs Sampling. Proceedings of the 43nd Annual Meeting of the Association for Computational Linguistics (ACL 2005), pp. 363-370. http://nlp.stanford.edu/~manning/papers/gibbscrf3.pdf

In more recent papers new implementations came out like the one depicted in the AllenNLP framework, described in the paper [Semi-supervised sequence tagging with bidirectional language models](https://www.semanticscholar.org/paper/Semi-supervised-sequence-tagging-with-bidirectional-Peters-Ammar/73e59cb556351961d1bdd4ab68cbbefc5662a9fc) that achieves state-of-the-art results in terms of F1-score accuracy for supervised tagging scores (+1% improvement to the current reference score), and by the way it is using a Gated Recurrent Unit (GRU) character encoder and CRF tagger at the end of the tagger model. A pre-trained model is provided as well [here](https://s3-us-west-2.amazonaws.com/allennlp/models/ner-model-2018.02.12.tar.gz). My questions are: 1) which is the current version of the ner module in the CoreNLP we are using and which is the current reference accuracy? Is it possibile to support the AlleNLP pre-trained ner model in the ner pipeline?
Thanks a lot.

**Comment by J38** [comment]
Hi, our CRF doesn't perform BIO or BIOES tagging, so it just tags with PERSON/LOCATION/ORGANIZATION/MISC or O.  Any consecutive sequence of PERSON* is considered a total entity.  In proper BIO or BIOES form, there are some cases in CoNLL 2003 where two entities are back to back, and separated by a B-PERSON tag in the middle.

So we have an F1 score of 89.82 if you use basic tagging instead of BIO or BIOES.  I think the basic tagging score is going to be higher than the proper BIO score (which is what is officially used in research papers), because errors of failing to place the B-PERSON tag for instance are ignored.

We are using an older technology which does not leverage deep learning, character embeddings, word embeddings, bi-lstm's, or Elmo. 

If you want to turn AllenNLP into a StanfordCoreNLP annotator, there's some Python code for creating a server, and there is some Java code for sending messages to that Python server.  No one has ever integrated AllenNLP.

Python project:

https://github.com/stanfordnlp/python-stanford-corenlp

Relevant Java class:

`edu.stanford.nlp.pipeline.GenericWebServiceAnnotator`

We may in the future release a Python-only version of a new dependency parser and pos-tagger.  It's possible a more modern NER system will be included in that.

**Comment by loretoparisi** [comment]
@J38 thanks a lot for your detailed explanation! 💯 

We are using the NER module as well as Regex one in the pipeline with good results, comparable to the AllenNLP biLSTM one. Also I think that having new DNN modules in Java is still a better solution, since it's easier to be integrated in enterprise and production ready system, while having in in Python, could trigger several problems of scalability.
Good to know about the `GenericWebServiceAnnotator`, that could be very useful as well, I did not know about it before!

**Comment by gui-li** [comment]
@J38 I am new to NER and I wanna know how people deal with the annation tagged by stanford NER?
You know, it does not support BIO format. Do people just simply convert continuous annoation to B-tag and I-tag in their result as baseline in their paper?
